File: micronaut-java11/src/main/java/demo/albertattard/uniquecode/CreateUniqueCodeRequest.java
package demo.albertattard.uniquecode;

import io.micronaut.core.annotation.Introspected;
import lombok.Data;

@Data
@Introspected
public class CreateUniqueCodeRequest {

    public static final int DEFAULT_LENGTH = 8;

    private String usedBy;
    private int length = DEFAULT_LENGTH;
    private String reference;
    private String description;
}

File: plain-go1_15/src/main/createUniqueCodeRequest.go
package main

type CreateUniqueCodeRequest struct {
	UsedBy      string `json:"usedBy"`
	Length      uint8  `json:"length"`
	Reference   string `json:"reference"`
	Description string `json:"description"`
}

func (request CreateUniqueCodeRequest) withDefaults() CreateUniqueCodeRequest {
	var length uint8 = 8
	if request.Length != 0 {
		length = request.Length
	}

	return CreateUniqueCodeRequest{
		UsedBy:      request.UsedBy,
		Length:      length,
		Reference:   request.Reference,
		Description: request.Description,
	}
}
